characters/watermelon.py

```diff
@@ -101,9 +101,24 @@ def fire_seed(self, model, seed_velocity):
             if self.health > 0:
                 self.health -= 10
 
+            state.offset = self.screen_shake(2, 3)
+
     def projectile_hit(self):
         self.health = min(self.health + 15, self.max_health)
 
+    def screen_shake(self, shake_count, shake_magnitude):
+        shake_maximum = shake_count * shake_magnitude
+        shake_vector = -1
+
+        for i in range(0, shake_count):
+            for x in range(0, shake_maximum, shake_magnitude):
+                yield (x * shake_vector, 0)
+            for x in range(shake_maximum, 0, shake_magnitude):
```

Comment:
I think you mean `for x in range(shake_maximum, 0, -shake_magnitude)`

`range` will only count backwards if step is negative

Reply:
D'oh.